asserts/ifacedecls.go

```diff
@@ -220,7 +222,256 @@ func compileAttributeConstraints(constraints interface{}) (*AttributeConstraints
 	return &AttributeConstraints{matcher: matcher}, nil
 }
 
+type fixedAttrMatcher struct {
+	result error
+}
+
+func (matcher fixedAttrMatcher) match(context string, v interface{}) error {
+	return matcher.result
+}
+
+var (
+	AlwaysMatchAttributes = &AttributeConstraints{matcher: fixedAttrMatcher{nil}}
+	NeverMatchAttributes  = &AttributeConstraints{matcher: fixedAttrMatcher{errors.New("not allowed, never matching")}}
+)
+
 // Check checks whether attrs don't match the constraints.
 func (c *AttributeConstraints) Check(attrs map[string]interface{}) error {
 	return c.matcher.match("", attrs)
 }
+
+func checkMapOrShortcut(context string, v interface{}) (m map[string]interface{}, shortcut bool, err error) {
+	switch x := v.(type) {
+	case map[string]interface{}:
+		return x, false, nil
+	case string:
+		switch x {
+		case "true":
+			return nil, true, nil
+		case "false":
+			return nil, false, nil
+		default:
+			return nil, false, fmt.Errorf("%s is not a map but is set to a shortcut that is not 'true' nor 'false': %q", context, x)
+		}
+	default:
+		return nil, false, fmt.Errorf("%s must be a map or one of the shortcuts 'true' or 'false'", context)
```

Comment:
This message seems nicer than the previous one, for both cases. Perhaps we can just remove both defaults and move that return to the end of the function.